### PR TITLE
API Docs initial RFC

### PR DESCRIPTION
**Pull request by mrkev**
Fun fact; theres 358 `#`s in this PR (:

Note: this includes mostly objects and CRUD-y stuff, but we'll need actions, login/auth, and attendance stuff too.

### CODE REVIEW

File: server/docs/api/Edges.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+All paginated routes return an object in the same format. If requesting a list
+of elements of `type T`, the response would be of type
+
+    type Response<T> {
+      edges: Array<Node<T>>
+      errors?: Array<Errors>
+    }
+
+    type Node<T> {
```
Comment (mrkev) [inline server/docs/api/Edges.md]: Oops, this shouldn't be called `Node<T>`, it should be an `Edge<T>`

**Comment by meganle** [comment]
Looks good! Will add necessary repo methods for each of the models.

File: server/docs/api/Courses.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+#### Response
+
+    200 OK -> Edges<Lectures>
+
+### Create a lecture
+#### Request
+
+    POST /v1/courses/${id}/lectures {
+      dateTime: string,
```
Comment (mrkev) [inline server/docs/api/Courses.md]: make this a number

File: server/docs/api/Courses.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
+
+# Courses
+
+## Courses
+
+    type Course {
+        id: string,
+        name: string,
+        term: string,
+    }
```
Comment (meganle) [inline server/docs/api/Courses.md]: Courses has some more fields like: subject (string) and catalogNum (number)

**Pull request merged**